sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/JavaClassLookupTransformProvider.java

```diff
@@ -488,8 +489,49 @@ public static AllowList everything() {
               AllowedClass.create("*", AllowedClass.WILDCARD, AllowedClass.WILDCARD)));
     }
 
+    static AllowList parseFromYamlStream(InputStream inputStream) {
+      Yaml yaml = new Yaml();
+      Map<Object, Object> config = yaml.load(inputStream);
+
+      if (config != null) {
+        String version = config.get("version") != null ? (String) config.get("version") : "";
+        List<AllowedClass> allowedClasses = new ArrayList<>();
+        if (config.get("allowedClasses") != null) {
+          allowedClasses =
+              ((List<Map<Object, Object>>) config.get("allowedClasses"))
+                  .stream()
+                      .map(
+                          data -> {
+                            String className = (String) data.get("className");
+                            if (className == null) {
+                              throw new IllegalArgumentException(
+                                  "Expected each entry in the allowlist to include the 'className'");
+                            }
+                            List<String> allowedBuilderMethods =
+                                (List<String>) data.get("allowedBuilderMethods");
+                            List<String> allowedConstructorMethods =
+                                (List<String>) data.get("allowedConstructorMethods");
+                            if (allowedBuilderMethods == null) {
+                              allowedBuilderMethods = new ArrayList<>();
+                            }
+                            if (allowedConstructorMethods == null) {
+                              allowedConstructorMethods = new ArrayList<>();
+                            }
+                            return AllowedClass.create(
+                                className, allowedBuilderMethods, allowedConstructorMethods);
+                          })
+                      .collect(Collectors.toList());
+        }
+        return AllowList.create(version, allowedClasses);
+      } else {
+        throw new IllegalArgumentException(
+            "Could not parse the provided YAML stream into a non-trivial AllowList");
+      }
+    }
+
     public abstract String getVersion();
 
+    @SuppressWarnings("mutable")
```

Comment:
I added these since I got some random warnings when compiling but seems like this didn't fully resolve the issue. So removing these suppressions for now.